cmd/zpool_influxdb/zpool_influxdb.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018-2020 Richard Elling
+ *
+ * This software is dual-licensed MIT and CDDL.
+ *
+ * The MIT License (MIT)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License Version 1.0 (CDDL-1.0).
+ * You can obtain a copy of the license from the top-level file
+ * "OPENSOLARIS.LICENSE" or at <http://opensource.org/licenses/CDDL-1.0>.
+ * You may not use this file except in compliance with the license.
+ *
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * CDDL HEADER END
+ */
+
+#ifndef	ZFS_ZPOOL_INFLUXDB_H
+#define	ZFS_ZPOOL_INFLUXDB_H
+
+#ifdef	__cplusplus
+extern "C" {
+#endif
+
+#define	SUPPORT_UINT64
+
+extern char *escape_string(char *);
+extern int print_scan_status(nvlist_t *, const char *);
+extern char *get_vdev_name(nvlist_t *, const char *);
+extern char *get_vdev_desc(nvlist_t *, const char *);
+extern int print_summary_stats(nvlist_t *, const char *, const char *);
+extern int print_vdev_latency_stats(nvlist_t *, const char *, const char *);
+extern int print_vdev_size_stats(nvlist_t *, const char *, const char *);
+extern int print_queue_stats(nvlist_t *, const char *, const char *);
+extern int print_top_level_vdev_stats(nvlist_t *, const char *);
+typedef int (*stat_printer_f)(nvlist_t *, const char *, const char *);
+extern int print_recursive_stats(stat_printer_f, nvlist_t *, const char *,
+	const char *, int);
+extern int print_stats(zpool_handle_t *, void *);
+extern void usage(char *);
+
+#ifdef	__cplusplus
+}
+#endif
+
+#endif /* ZFS_ZPOOL_INFLUXDB_H */
```

Comment:
Can't we do without this header?

Reply:
I'm using it as a convenient place to define SUPPORT_UINT64. But perhaps it is better to reverse that logic and allow override for not supporting uint64. Thoughts?

Reply:
I think either way can be done without a header.

If you see this being a regularly used configuration option, I would add something to ./configure for it instead. Something that detects the influxdb version by default and can be outsmarted by `--with-influxdb=2` or the like.

This doesn't actually have a build dependency on influxdb, so falling back to the newest version if not present would seem appropriate. With that in mind, inverting the logic to be an opt-in for compat with the older version makes sense.

Reply:
The change for unsigned support actually is several years old now. When using telegraf, which is the preferred method, unsigned ints are handled properly by telegraf and we don't really need a recompile. I'd like to treat this as a "don't go back to the ice age"